libs/algorithms/include/hpx/parallel/algorithms/reduce.hpp

```diff
@@ -82,12 +83,17 @@ namespace hpx { namespace parallel { inline namespace v1 {
         };
         /// \endcond
         // Non Segmented Reduce
+        //
+        // clang-format off
         template <typename ExPolicy, typename FwdIterB, typename FwdIterE,
-            typename T, typename F>
-        inline typename std::enable_if<
-            execution::is_execution_policy<ExPolicy>::value,
-            typename util::detail::algorithm_result<ExPolicy, T>::type>::type
-        reduce_(ExPolicy&& policy, FwdIterB first, FwdIterE last, T init, F&& f,
+            typename T, typename F,
+            HPX_CONCEPT_REQUIRES_(
+                execution::is_execution_policy<ExPolicy>::value &&
+                hpx::traits::is_sentinel_for<FwdIterE, FwdIterB>::value
+            )>
+        // clang-format on
+        typename util::detail::algorithm_result<ExPolicy, T>::type reduce_(
```

Comment:
@hkaiser is the `inline` supposed to be gone ?

Reply:
The `inline` does not make a big difference on template functions (those are `inline` implicitly). Removing it only removes a hint to the compiler to attempt inlining this function. But that's what it will do in any case.

Reply:
But just out of curiosity, isn't removing all those hints increasing our compile time ?

Reply:
I wouldn't expect any measurable impact on the compilation process. If anything, compile times should go down as the compiler has to inline less code. I have no proof, however. In general I'm fine with adding the `inline` back, even if I think we don't do that consistently anyways.